src/client.lisp

```diff
@@ -31,6 +31,12 @@
 
 (in-package #:rpcq)
 
+(defstruct client-auth-config
+  "Holds the ZeroMQ Curve configuration for a client socket."
+  (client-secret-key nil :type (or null string))
```

Comment:
can these really be null? can they be RO?

Reply:
No. Originally I had an `error` form as the default, but then decided I was maybe being too uptight and switched to defaulting to `nil` to make repl testing easier if you just want a partially constructed thing. Happy to switch back to requiring non-nil here. readonly makes sense as well.